201-front-door-rate-limiting/azuredeploy.json

```diff
@@ -32,26 +43,48 @@
             "metadata": {
                 "description": "Defines rate limit thresold."
             }
+        },
+        "rateLimitAction": {
+            "type": "string",
+            "allowedValues": [
+                "Allow",
+                "Block",
+                "Log"
+            ],
+            "defaultValue": "Log",
+            "metadata": {
+                "description": "Type of Action based on the match filter. Must be Allow, Block or Log."
+            }
+        },
+        "ipMatch": {
+            "type": "string",
+            "metadata": {
+                "description": "The operator to be matched."
+            }
         }
     },
     "variables": {
-        "frontdoorref": "[resourceId('Microsoft.Network/frontDoors', parameters('frontDoorName'))]",
-        "wafpolicyref": "[resourceId('Microsoft.Network/FrontDoorWebApplicationFirewallPolicies', parameters('wafPolicyName'))]",
-        "frontdoorLocation": "global"
+        "frontdoorLocation": "global",
+        "frontdoorSuffix": ".azurefd.net"
```

Comment:
use reference

Reply:
The resource cannot make a reference to itself while is being created.